pkg/mirror/service.go

```diff
@@ -190,10 +191,29 @@ func (p *pullThroughMirror) upstreamSha256Sums(ctx context.Context, provider *co
 	return p.upstream.shaSums(ctx, providerUpstream)
 }
 
-func NewPullThroughMirror(s Storage, c Copier) Service {
+func NewPullThroughMirror(s Storage, c Copier, cacheConfig CacheConfig) Service {
 	remoteServiceDiscovery := discovery.NewRemoteServiceDiscovery(http.DefaultClient)
+
+	// Create the base upstream which consumes the remote API
+	var upstream upstreamProvider = newUpstreamProviderRegistry(remoteServiceDiscovery)
+
+	// Wrap with cache if enabled
+	if cacheConfig.Enabled {
+		cachedUpstream, err := newCachedUpstreamProvider(upstream, cacheConfig)
+		if err != nil {
+			// Graceful fallback: log warning and continue without cache
+			slog.Warn("failed to initialize cache, continuing without caching",
+				slog.String("error", err.Error()))
```

Comment:
I think it would be better to propagate the error up the caller chain. If there is an error, this means there is probably a bug or misconfiguration and I think we should probably terminate the process